params/protocol_params.go

```diff
@@ -279,11 +302,15 @@ func CalculateGasWithStateScaling(stateSize, contractSize *big.Int, baseRate uin
 	return new(big.Int).Div(num, den).Uint64()
 }
 
-func CalculateCoinbaseValueWithLockup(value *big.Int, lockupByte uint8) *big.Int {
-	if lockupByte == 0 {
+func CalculateCoinbaseValueWithLockup(value *big.Int, lockupByte uint8, blockNumber uint64) *big.Int {
+	if lockupByte == 0 || blockNumber < 2*BlocksPerMonth {
 		return value
+	} else {
+		rewardsMultiple, _ := CalculateLockupByteRewardsMultiple(lockupByte, blockNumber)
+		rewardValue := new(big.Int).Mul(value, rewardsMultiple)
+		rewardValue = new(big.Int).Div(rewardValue, big.NewInt(10000))
```

Comment:
I'm confused here. Suppose lockupByte = 1 and year = 1. rewardsMultplie is 3500 correct? (100 * 3500) / 10000 = 35
So they lose money?

Reply:
there is an addition of the original value, this is just the interest part, its missing, added it locally, will push. I am changing this to be a continuous curve instead of discrete values as well, so this function is getting modified